packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/gateway/L1GatewayRouter.sol

```diff
@@ -120,14 +121,43 @@ contract L1GatewayRouter is WhitelistConsumer, L1ArbitrumMessenger, GatewayRoute
         bytes memory data =
             abi.encodeWithSelector(L2GatewayRouter.setGateway.selector, _token, _gateway);
 
-        return sendTxToL2(msg.sender, 0, _maxSubmissionCost, _maxGas, _gasPriceBid, data);
+        return sendTxToL2(_creditBackAddress, 0, _maxSubmissionCost, _maxGas, _gasPriceBid, data);
     }
 
+    /**
+     * @notice Allows L1 Token contract to trustlessly register its gateway. (other setGateway method allows excess eth recovery from _maxSubmissionCost and is recommended)
+
+     * @param _gateway l1 gateway address
+     * @param _maxGas max gas for L2 retryable exrecution 
+     * @param _gasPriceBid gas price for L2 retryable ticket 
+     * @param  _maxSubmissionCost base submission cost  L2 retryable tick3et 
+     * @return Retryable ticket ID
+     */
     function setGateway(
         address _gateway,
         uint256 _maxGas,
         uint256 _gasPriceBid,
         uint256 _maxSubmissionCost
+    ) external payable returns (uint256) {
+        this.setGateway(_gateway, _maxGas, _gasPriceBid, _maxSubmissionCost, msg.sender);
```

Comment:
Should we instead `return setGateway(_gateway, _maxGas, _gasPriceBid, _maxSubmissionCost, msg.sender);`